R/extract.R

```diff
@@ -68,13 +68,26 @@ filter_chunk_end_positions <- function(starts, ends) {
   }
 
   positions <- sort(c(starts = starts, ends = ends))
-  code_start_indexes <- which(grepl("starts", names(positions)))
+  code_start_indexes <- grep("starts", names(positions))
   code_ends <- positions[1 + code_start_indexes]
 
   stopifnot(all(grepl("ends", names(code_ends))))
   code_ends
 }
 
+defines_knitr_engine <- function(line) {
+  # "scala|python|bash|..."
+  engine_choices <- paste(names(knitr::knit_engines$get()), collapse = "|")
+
+  # Looks for {some_engine}, {some_engine label, ...} or {some_engine, ...}
+  bare_engine_pattern <- paste0("\\{(", engine_choices, ")[, }]")
```

Comment:
please construct this pattern using rex, it would be something like

```r
rex::rex("{", or(names(knitr::knit_engines$get())), one_of("}", " ", ","))
```

Reply:
No problem. I've modified that and pushed. Hadn't used `rex` before.